File: app/src/main/java/com/example/prashant/databinding/data/ContactDao.java
package com.example.prashant.databinding.data;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import static android.arch.persistence.room.OnConflictStrategy.REPLACE;

@Dao
public interface ContactDao {
    @Insert(onConflict = REPLACE)
    void save(Contact contact);

    @Query("SELECT * FROM contact WHERE id = :id")
    LiveData<Contact> load(String id);

    @Query("SELECT * FROM contact")
    LiveData<List<Contact>> loadAll();

    @Query("DELETE FROM contact WHERE id = :id")
    void delete(String id);

    @Update
    void update(Contact contact);
}

File: app/src/main/java/com/example/prashant/databinding/data/ContactDao.kt
package com.example.prashant.databinding.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy.REPLACE
import androidx.room.Query
import androidx.room.Update

@Dao
interface ContactDao {
    @Insert(onConflict = REPLACE)
    fun save(contact: Contact)

    @Query("SELECT * FROM contact WHERE id = :id")
    fun load(id: Int): LiveData<Contact>

    @Query("SELECT * FROM contact")
    fun loadAll(): LiveData<List<Contact>>

    @Query("DELETE FROM contact WHERE id = :id")
    fun delete(id: Int)

    @Update
    fun update(contact: Contact)
}